diagnostics/runoff_sensitivities/runoff_sensitivities_diag.py

```diff
@@ -0,0 +1,1019 @@
+    raise ValueError('water budget is not closed more than 80% of ' + str(nb) + 'basins')
+
+ 
+
+##############################################################################
+##  3) calculate runoff sensitivities  ##########
+##############################################################################
+# initialization
+psens_model = np.full((nb, 3),np.nan)
+tsens_model = np.full((nb, 3),np.nan)
+intsens_model = np.full((nb, 3),np.nan)
+r2_mrro_wy_int = np.full((nb),np.nan)
+r2_model_int = np.full((nb),np.nan)
+r2_model_pred = np.full((nb),np.nan)
+r2_model_pred_int = np.full((nb),np.nan)
+
+# water year average (Oct-Sep water year for every basin)
+# after 5-year average, the definition of differnet water year for each basin does not matter much
+mw=5 # 5-year moving window
+for b in range(nb):
+    # specify start and end indices
+
+    # smoothing (5-year moving average)
+    mrrobym = moving_avg(mrrob_wy[:,b],mw)
+    prbym = moving_avg(prb_wy[:,b],mw)
+    tasbym = moving_avg(tasb_wy[:,b],mw)
+
+    # get anomalies
+    amrrobym = mrrobym - np.nanmean(mrrobym)
+    aprbym = prbym - np.nanmean(prbym)
+    atasbym = tasbym - np.nanmean(tasbym)
+
+    # get % anomalies
+    amrrobym_pct = amrrobym / np.nanmean(mrrobym) * 100
+    aprbym_pct = aprbym / np.nanmean(prbym) * 100
+
+    # calculate runoff sensitivities using def runoff_sens_reg(r, p, t):
+    a1, a2, b1, b2, c1, c2, r2 = runoff_sens_reg(amrrobym_pct, aprbym_pct, atasbym, mw=5, alpha=0.1)
+    psens_model[b,0] = a1
+    psens_model[b,1:3] = a2
+    tsens_model[b,0] = b1
+    tsens_model[b,1:3] = b2
+    intsens_model[b,0] = c1
+    intsens_model[b,1:3] = c2
+    r2_mrro_wy_int[b] = r2
+
+    # prediction and resulting R2 (training accuracy)
+    amrrobym_pct_pred_model = a1 * aprbym_pct + b1 * atasbym
+    amrrobym_pct_pred_int_model = a1 * aprbym_pct + b1 * atasbym + c1 * atasbym * aprbym_pct
+    corr = corr_nan(amrrobym_pct_pred_model, amrrobym_pct_pred_int_model)
+    r2_model_int[b] = corr**2
+    corr = corr_nan(amrrobym_pct_pred_model, amrrobym_pct)
+    r2_model_pred[b] = corr**2
+    corr = corr_nan(amrrobym_pct_pred_int_model, amrrobym_pct)
+    r2_model_pred_int[b] = corr**2
+
+# display warning if the interaction term is affecting the regression models
+# role of interaction term (P*T) is usually negligible, consistent to Lehner et al. 2019
+condition=np.asarray(r2_model_int<0.8)
+# condition=np.asarray(np.abs(r2_model_pred_int-r2_model_pred)>0.2)
+for b in range(b):
+    if condition[b]:
+        warning_message = f'Warning: Interaction term may matter for model + basin #{b+1} (r={r2_model_int[b]:.2f})'
+        warnings.warn(warning_message)
+
+
+
+#####################################################
+### 4) Saving output data: ##########################
+#####################################################
+## save runoff sensitivity data in netCDF4 file
+out_path = "{WK_DIR}/model/netCDF/runoff_sensitivities_{CASENAME}.nc".format(**os.environ)

```

Comment:
replace 'WK_DIR' with 'WORK_DIR'

Reply:
@bitterbark @hanjunkim0617 was it changed to WORKING_DIR? Change it to WORK_DIR.